src/api/route-services/pipelines.js

```diff
@@ -0,0 +1,196 @@
+            Path: `/apis/batch/v1/namespaces/${namespace}/jobs`,
+            QueryParameters: {
+              fieldSelector: [
+                'status.successful=1',
+              ],
+            },
+          },
+          Next: 'LaunchKubernetesJobIfNotExists',
+        },
+        LaunchKubernetesJobIfNotExists: {
+          Type: 'Task',
+          Comment: 'Attempts to create a Kubernetes Job for the pipeline server. Will swallow a 409 (already exists) error.',
+          Resource: 'arn:aws:states:::eks:call',
+          Parameters: {
+            ClusterName: name,
+            CertificateAuthority: certAuthority,
+            Endpoint: endpoint,
+            Method: 'POST',
+            Path: `/apis/batch/v1/namespaces/${namespace}/jobs`,
+            RequestBody: {
+              apiVersion: 'batch/v1',
+              kind: 'Job',
+              metadata: {
+                name: `remoter-server-${experimentId}`,
+              },
+              spec: {
+                template: {
+                  metadata: {
+                    name: `remoter-server-${experimentId}`,
+                  },
+                  spec: {
+                    containers: [
+                      {
+                        name: 'remoter-server',
+                        image,
+                      },
+                    ],
+                    restartPolicy: 'Never',
+                  },
+                },
+              },
+            },
+          },
+          Retry: [
+            {
+              ErrorEquals: ['EKS.409'],
+              IntervalSeconds: 1,
+              BackoffRate: 2.0,
+              MaxAttempts: 2,
+            },
+          ],
+          Catch: [
+            {
+              ErrorEquals: ['EKS.409'],
+              ResultPath: '$.error-info',
+              Next: 'Wait',
+            },
+          ],
+          Next: 'Wait',
+        },
+        Wait: {
+          Type: 'Wait',
+          Seconds: 5,
+          End: true,
+        },
+      },
+    };
+  }
+
+
+  async create(experimentId) {
+    const {
+      roleArn, accountId,
+    } = await this.getClusterInformation();
+
+    const { clusterEnv, sandboxId } = config;
+
+    const pipelineHash = crypto
+      .createHash('sha1')
+      .update(`${experimentId}-${sandboxId}`)
+      .digest('hex');
+
+    const params = {
+      name: `biomage-pipeline-${pipelineHash}`,
+      roleArn,
+      definition: JSON.stringify(await this.generatePipeline(experimentId)),
+      loggingConfiguration: {
+        level: 'OFF',
+      },
+      tags: [
+        {
+          key: 'experimentId',
+          value: experimentId,
+        },
+        {
+          key: 'clusterEnv',
+          value: clusterEnv,
+        },
+        {
+          key: 'sandboxId',
+          value: sandboxId,
+        },
+      ],
+      type: 'STANDARD',
+    };
+
+    try {
+      const { stateMachineArn } = await this.stepFunctions.createStateMachine(params).promise();
```

Comment:
where do you define the actual steps that the step machine will be running with?

Reply:
Those are defined in `generatePipeline`. For now, the state machine is static and is hard-coded. This will have to be changed for local development, because the steps will have to be different (running Lambdas or something similar instead of Kubernetes jobs), so they will need to be refactored as part of that ticket when I am more sure about what the exact use case will be.

Reply:
they will also need to be different because the users need to be able to exclude certain filters from the pipeline. Ok, if the plan is to update them later, sounds good.

Reply:
Correct, I know. Plan is to have some sort of map-reduce function to take the config, construct a new pipeline stepwise (with different step configurations for local and AWS development, as discussed in the morning), and put it together. This is just a placeholder for now.